v/vm.c

```diff
@@ -107,6 +117,57 @@ void printhex(uint64_t x)
   cputstring(str);
 }
 
+// for testing purpose only alloc
+static pte_t alloc()
+{
+  if (freelist_head == 0) {
+    return OUT_OF_MEMORY;
+  }
+  pte_t res = freelist_head->addr;
+  if (freelist_head != freelist_tail)
+    freelist_head = freelist_head->next;
+  else
+    freelist_head = freelist_tail = 0;
+  return res;
+}
+
```

Comment:
These seem like substantial changes to how the VM works. @farzamgl is all this necessary to support the 2MB pages?

Reply:
This code generalizes the page-table generation so we can have more than 2MB space. It's not that different except the way that the mappings are done, and there is no page copying.
In the original code, there is only 1 entry in L2 page-table for each user and kernel code.

Reply:
In the vt/*, there will be user code copying and remapping when a timer interrupt is taken.